README.md

```diff
@@ -1,2 +1,178 @@
+</proxy>
+```
+
+# How to consume the dependency
+
+   ```
+   <dependency>
+      <groupId>com.github.znsio</groupId>
+      <artifactId>ReportPortalIntegration</artifactId>
+      <version>x.x.x</version>
+   </dependency>
+   ```
+
+# How to configure ReportPortal in your automation framework
+
+1. **Listener Configuration**
+    1. **Approach-1 Listener in TestNG XML file:** Add the following line to configure ReportPortal listener in the xml
+       file where you have configured your testng suite
+        1. **Without overloading properties**
+           ```
+           <listener class-name="com.epam.reportportal.testng.ReportPortalTestNGListener"/>
+           ```
+        2. **With overloading properties**
+           ```
+           <listener class-name="com.znsio.rpi.listener.ReportPortalListener"/>
+           ```
+    2. **Approach-2 Listener in Maven Surefire Plugin:** Add the following snippet to configure ReportPortal listener in
+       the `pom.xml` file where you have configured your `maven-surefire-plugin`
+        1. **Without overloading properties**
+           ```
+           <property>
+              <name>listener</name>
+              <value>com.epam.reportportal.testng.ReportPortalTestNGListener</value>
+           </property>
+           ```
+        2. **With overloading properties**
+           ```
+           <property>
+              <name>listener</name>
+              <value>com.znsio.rpi.listener.ReportPortalListener</value>
+           </property>
+           ```
+    2. **Approach-3:** Use the following snippet to configure ReportPortal listener when creating TestNG file
+       programmatically
+        1. **Without overloading properties**
+           ```
+           ReportPortalTestNGListener reportPortalTestNGListener = new ReportPortalTestNGListener();
+           testNG.addListener(reportPortalTestNGListener);
+           ```
+        2. **With overloading properties**
+           ```
+           ReportPortalListener reportPortalListener = new ReportPortalListener();
+           testNG.addListener(reportPortalListener);
+           ```
+    3. For more details on listener configuration,
+       refer [reportportal-agent-java-testNG](https://github.com/reportportal/agent-java-testNG)
+2. **Properties file:** Create a `reportportal.properties` file either in `src/main/resources/` or
+   in `src/test/resources` directory in your automation framework with the following mandatory attributes
+   ```
+   rp.endpoint = <Your Endpoint>
+   rp.uuid = <Your UDID>
+   rp.launch = <Your Launch Name>
+   rp.project = <Your Project Name>
+   ```
+3. **ReportPortal Attributes and Properties Overlaoding:** Use the `ReportPortalPropertiesOverloader` class defined
+   under `src/main/java/com/znsio/rpi/properties` path to change any attributes/property value for ReportPortal at
+   runtime
+4. **Launch attributes related to CI (Pipeline) execution**: The `setPipelineAttributes` method
```

Comment:
Why is this required?

Reply:
Here we're explaining how tutility is setting the pipeline attributes and how to configure them in your consumer framework to get correct values in ReportPortal